data_model/src/isi.rs

```diff
@@ -135,124 +132,27 @@ pub mod model {
-                Revoke(revoke_box) => revoke_box.len(),
-                ExecuteTrigger(execute_trigger) => execute_trigger.len(),
-                SetParameter(set_parameter) => set_parameter.len(),
-                NewParameter(new_parameter) => new_parameter.len(),
-                Upgrade(upgrade_box) => upgrade_box.len(),
-            }
-        }
-    }
+    impl Instruction for InstructionBox {}
 
-    impl Instruction for SetKeyValueBox {
-        fn len(&self) -> usize {
-            self.object_id.len() + self.key.len() + self.value.len() + 1
-        }
-    }
-    impl Instruction for RemoveKeyValueBox {
-        fn len(&self) -> usize {
-            self.object_id.len() + self.key.len() + 1
-        }
-    }
-    impl Instruction for RegisterBox {
-        fn len(&self) -> usize {
-            self.object.len() + 1
-        }
-    }
-    impl Instruction for UnregisterBox {
-        fn len(&self) -> usize {
-            self.object_id.len() + 1
-        }
-    }
-    impl Instruction for MintBox {
-        fn len(&self) -> usize {
-            self.destination_id.len() + self.object.len() + 1
-        }
-    }
-    impl Instruction for BurnBox {
-        fn len(&self) -> usize {
-            self.destination_id.len() + self.object.len() + 1
-        }
-    }
-    impl Instruction for TransferBox {
-        fn len(&self) -> usize {
-            self.destination_id.len() + self.object.len() + self.source_id.len() + 1
-        }
-    }
-    impl Instruction for GrantBox {
-        fn len(&self) -> usize {
-            self.object.len() + self.destination_id.len() + 1
-        }
-    }
-    impl Instruction for RevokeBox {
-        fn len(&self) -> usize {
-            self.object.len() + self.destination_id.len() + 1
-        }
-    }
-    impl Instruction for SetParameterBox {
-        fn len(&self) -> usize {
-            self.parameter.len() + 1
-        }
-    }
-    impl Instruction for NewParameterBox {
-        fn len(&self) -> usize {
-            self.parameter.len() + 1
-        }
-    }
-    impl Instruction for UpgradeBox {
-        fn len(&self) -> usize {
-            self.object.len() + 1
-        }
-    }
-    impl Instruction for ExecuteTriggerBox {
-        fn len(&self) -> usize {
-            1
-        }
-    }
-    impl Instruction for FailBox {
-        fn len(&self) -> usize {
-            1
-        }
-    }
+    impl Instruction for SetKeyValueBox {}
+    impl Instruction for RemoveKeyValueBox {}
+    impl Instruction for RegisterBox {}
+    impl Instruction for UnregisterBox {}
+    impl Instruction for MintBox {}
+    impl Instruction for BurnBox {}
+    impl Instruction for TransferBox {}
+    impl Instruction for GrantBox {}
+    impl Instruction for RevokeBox {}
+    impl Instruction for SetParameterBox {}
+    impl Instruction for NewParameterBox {}
+    impl Instruction for UpgradeBox {}
+    impl Instruction for ExecuteTriggerBox {}
+    impl Instruction for FailBox {}
```

Comment:
Probably useful to use `syntactic_for` here.

Reply:
interesting crate, after my update I don't think it's useful in this case